Dockerfile

```diff
@@ -5,18 +5,18 @@ FROM chef AS planner
 COPY . .
 # Mitigate rustup 1.28.0's new behavior https://blog.rust-lang.org/2025/03/02/Rustup-1.28.0.html
 # Supposedly, it will be rolled back in the next patch release https://github.com/rust-lang/rustup/pull/4214
-RUN rustup show active-toolchain || rustup toolchain install
 RUN cargo chef prepare --recipe-path recipe.json
 
 FROM chef AS builder
 COPY --from=planner /anvil-zksync/recipe.json recipe.json
+COPY rust-toolchain.toml rust-toolchain.toml
 # Build dependencies - this is the caching Docker layer
 RUN cargo chef cook --release --recipe-path recipe.json
 # Build application
 COPY . .
 # Mitigate rustup 1.28.0's new behavior https://blog.rust-lang.org/2025/03/02/Rustup-1.28.0.html
 # Supposedly, it will be rolled back in the next patch release https://github.com/rust-lang/rustup/pull/4214
-RUN rustup show active-toolchain || rustup toolchain install
+RUN rustup install nightly-2025-03-19 && rustup default nightly-2025-03-19
```

Comment:
Hmm, why is it necessary to specify the version here? Shouldn't `nightly-2025-03-19` get installed automatically from `rust-toolchain.toml`?